phpBB/phpbb/captcha/plugins/qa.php

```diff
@@ -224,6 +224,11 @@ function get_template()
 
 		if ($this->is_solved() || empty($this->question_text) || !count($this->question_ids))
 		{
+			// don't log an error if the captcha has been solved
```

Comment:
Maybe it will be better to move this check into a separate if branch?
Result:
```
		if ($this->is_solved())
		{
			return;
		}
		else if (empty($this->question_text) || !count($this->question_ids))
		{
			/** @var \phpbb\log\log_interface $phpbb_log */
			$phpbb_log->add('critical', $user->data['user_id'], $user->ip, 'LOG_ERROR_CAPTCHA', time(), array($user->lang('CONFIRM_QUESTION_MISSING')));
			return false;
		}
		else
		{
			$template->assign_vars(array(
				'QA_CONFIRM_QUESTION'		=> $this->question_text,
				'QA_CONFIRM_ID'				=> $this->confirm_id,
				'S_CONFIRM_CODE'			=> true,
				'S_TYPE'					=> $this->type,
			));

			return 'captcha_qa.html';
		}
```